traits/ctraits.c

```diff
@@ -4627,15 +4653,6 @@ set_trait_modify_delegate_flag(
     return set_trait_flag(trait, TRAIT_MODIFY_DELEGATE, value);
 }
 
-/*-----------------------------------------------------------------------------
-|  Returns the current object_id_test flag value:
-+----------------------------------------------------------------------------*/
-
-static PyObject *
-get_trait_object_id_test_flag(trait_object *trait, void *closure)
-{
-    return get_trait_flag(trait, TRAIT_OBJECT_ID_TEST);
-}
```

Comment:
Considering this was introduced in #666, should this (and `get_trait_no_value_test_flag`) be kept? The `comparison_mode` property in `ctrait.CTrait` can be implemented using these properties.

Reply:
They *could* be kept as static functions in `ctraits.c`, not exposed to CPython, if that makes the C implementation easier / more readable. But I definitely think they should be removed from the `cTrait` interface, and I don't think it makes sense to keep them on the C implementation front, either. Conceptually, this is a single piece of trait metadata that we're setting and getting, even if it happens to be represented using two flag bits.

Reply:
I'd actually be inclined to rework the flag constants to make the C implementation more direct / straightforward. Instead of the flags:
```
#define TRAIT_OBJECT_ID_TEST 0x00000004U
#define TRAIT_NO_VALUE_TEST 0x00000100U
```
you could define constants
```
#define TRAIT_NO_COMPARE 0x00000100U
#define TRAIT_IDENTITY_COMPARE 0x000000004U
#define TRAIT_EQUALITY_COMPARE 0x00000000U
#define TRAIT_COMPARE_MASK 0x00000104U
```

and then the `comparison_mode` getter simply does `flags & TRAIT_COMPARE_MASK` and compares with each of the three possibilities in turn, while the `comparison_mode` setter clears the relevant bits using `flags &= ~TRAIT_COMPARE_MASK` and then ors in the appropriate bits.

But that doesn't have to happen in this PR.